### ISSUE TITLE
Development Store Installation - Odd Error

### ISSUE THREAD

**Issue by MCKLtech** [ohmybrew/laravel-shopify#227]
I've created a fresh project and I'm running on Heroku.

When I begin the install procedure from http://blooming-depths-XXXX.herokuapp.com/ by entering my development site URL, the installation proceeds as normal i.e. Warned about a development store, accept accesses and then done.

However, if I start the install from my partner dashboard e.g. 'Install on Development Store', the installation starts but errors out, in the development store, as:

**The page you’re looking for couldn’t be found**

And the App is not installed or present under 'Apps'. A review of the Network tools shows:

https://app-context-1.myshopify.com/admin/apps/XXX/?hmac=XXX&shop=dev-shop-1234.myshopify.com&timestamp=1556286404

with a **404 Not Found** error.

I'm not sure where to begin on this. I've compared to an App that is working and all the .env variables are set up, in addition to the views. There is nothing in my Heroku logs but I can see the requests coming in.

**Edit:**

_**This error does not occur**_ when I downgrade to **ohmybrew/laravel-shopify:6.0** as per the other app that is working. Hence, it seems to only occur on 7.0. From looking at commits, would this possibly be related to session handling?

**Comment by ohmybrew** [comment]
Thanks, I'll look into this. The session handling hasn't changed exactly, just moved into its own file. I'll give this a go and see whats up.

**Comment by MCKLtech** [comment]
Just to add, with the newest commit, this is still happening.

**Comment by MCKLtech** [comment]
https://pasteboard.co/IgEnv3x.png

**Process**

- Attempt install of Shopify App on Development Store from Partners Panel. As soon as it redirects, this happens.

Appears to only occur on re-installs.

**Comment by ohmybrew** [comment]
OK so install from dashboard, not the direct login URL correct?On May 27, 2019 12:17, MCKLtech <notifications@github.com> wrote:https://pasteboard.co/IgEnv3x.png
Process
Attempt install of Shopify App on Development Store from Partners Panel. As soon as it redirects, this happens.
Appears to only occur on re-installs.

**Comment by MCKLtech** [comment]
Version 7.0 & Version 6.0

Process One

- Install from OhMyBrew 'Login' screen using Development Store URL
- Install is successful
- Remove App via Shopify Admin Panel
- Try install App again from Login Screen. Automatically redirects to Shopify Admin Panel and states 'The page you’re looking for couldn’t be found'
- Clear Cookies on all devices.
- Same result.

Process Two

- Open Shopify Partners Admin Panel
- Install App to Development Store
- Automatically redirects to Shopify Store Admin Panel and states 'The page you’re looking for couldn’t be found'

Notably, once the App is installed I'm never prompted again to accept the App installation e.g. The unlisted warning. It seems the App isn't being flagged as uninstalled which may explain why I'm brought to a page not found as it may be looking for the App I just uninstalled?

**Comment by ohmybrew** [comment]
Are you using an uninstall webhook? The one that comes with the package (disabled by default I believe), will watch for whenever shopify sends the webhook and will clear the access token and trash (not delete) the shop.If thats not in place, I imagine there's residule auth stuff there which causes it to skip the author process and think its installed.On May 27, 2019 12:32, MCKLtech <notifications@github.com> wrote:Version 7.0 & Version 6.0

Process One

Install from OhMyBrew 'Login' screen using Development Store URLInstall is successfulRemove App via Shopify Admin PanelTry install App again from Login Screen. Automatically redirects to Shopify Admin Panel and states 'The page you’re looking for couldn’t be found'Clear Cookies on all devices.Same result.

Process Two

Open Shopify Partners Admin PanelInstall App to Development StoreAutomatically redirects to Shopify Store Admin Panel and states 'The page you’re looking for couldn’t be found'

Notably, once the App is installed I'm never prompted again to accept the App installation e.g. The unlisted warning. It seems the App isn't being flagged as uninstalled which may explain why I'm brought to a page not found as it may be looking for the App I just uninstalled?

**Comment by MCKLtech** [comment]
I found the issue and it's entirely my fault.

In case anyone faces this, pay close attention to how you uninstall your App via webhooks:

```
 [
            'topic' => 'app/uninstalled',
            'address' => env('APP_URL').'/webhook/app-uninstalled'
 ]
```

My mistake was the omission of the **/webhook/** in the overall URL hence my App was never being uninstalled properly.

Thanks, OhMyBrew and sorry for any time wasted :/

**Comment by ohmybrew** [comment]
No worries! Never a waste of time 😃. Sadly I don't know of a way right now to get around it. The code needs to know it was uninstalled, and clearing the token and trashing the shop is one of those two ways. Without it, the code sees the shop come in, sees it in the database,  sees it has a token and assumes all is well.. Theres no way to check otherwise. On May 27, 2019 12:50, MCKLtech <notifications@github.com> wrote:Closed #227.
